src/NeoSharp.VM/Types/InteropService.cs

```diff
@@ -88,84 +77,70 @@ public InteropService()
         /// <summary>
         /// Register method
         /// </summary>
-        /// <param name="method">Method name</param>
-        /// <param name="synonymous">Synonymous</param>
+        /// <param name="name">Method name</param>
         /// <param name="handler">Method delegate</param>
         /// <param name="gas">Gas</param>
-        public void Register(string method, string synonymous, InteropServiceEntry.delHandler handler, uint gas = 1)
+        public void Register(string name, Func<IExecutionEngine, bool> handler, uint gas = 1)
         {
-            _entries[method.ToInteropMethodHash()] = new InteropServiceEntry(method, handler, gas);
-            _entries[synonymous.ToInteropMethodHash()] = new InteropServiceEntry(synonymous, handler, gas);
+            Register(name, null, handler, gas);
         }
 
         /// <summary>
         /// Register method
         /// </summary>
-        /// <param name="method">Method name</param>
+        /// <param name="name">Method name</param>
+        /// <param name="alias">Alias</param>
         /// <param name="handler">Method delegate</param>
         /// <param name="gas">Gas</param>
-        public void Register(string method, InteropServiceEntry.delHandler handler, uint gas = 1)
+        public void Register(string name, string alias, Func<IExecutionEngine, bool> handler, uint gas = 1)
         {
-            _entries[method.ToInteropMethodHash()] = new InteropServiceEntry(method, handler, gas);
-        }
+            var entry = new InteropServiceEntry(name, GetMethodHash(name), handler, gas);
+            _entries.Add(entry.MethodHash, entry);
 
-        /// <summary>
-        /// Clear entries
-        /// </summary>
-        public void Clear()
-        {
-            _entries.Clear();
+            if (alias == null) return;
+
+            entry = new InteropServiceEntry(alias, GetMethodHash(alias), handler, gas);
+            _entries.Add(entry.MethodHash, entry);
         }
 
         /// <summary>
-        /// Invoke
+        /// Invoke method
         /// </summary>
-        /// <param name="method">Method name</param>
+        /// <param name="method">Method</param>
         /// <param name="engine">Execution engine</param>
-        /// <returns>Return false if something wrong</returns>
+        /// <returns>Return false if something is wrong</returns>
         public bool Invoke(byte[] method, IExecutionEngine engine)
         {
-            uint hash = method.Length == 4 ? BitConverter.ToUInt32(method, 0)
-                 : Encoding.ASCII.GetString(method).ToInteropMethodHash();
+            var hash = method.Length == 4
+                ? BitConverter.ToUInt32(method, 0)
+                : GetMethodHash(Encoding.ASCII.GetString(method));
+
+            void InvokeOnSysCall(SysCallResult result) => OnSysCall?.Invoke(this, new SysCallArgs(engine, GetMethodName(hash), hash, result));
```

Comment:
This is very weird xD

Reply:
Yep, I used to use `Func` or `Action` delegates instead but VS converts them into local methods all the time.